util/include/cppmicroservices/util/BundleElfFile.h

```diff
@@ -144,62 +145,67 @@ class BundleElfFile
               sizeof dynamicSecEntry);
     }
 
-    // parse the .dynsym section
-    Shdr* dynsymHdr = this->FindSectionHeader(SHT_DYNSYM);
-    if (dynsymHdr == nullptr) {
-      throw InvalidElfException("ELF .dynsym section header missing");
-    }
-    strTab = this->GetStringTable(fs, dynsymHdr);
-    std::size_t sectionSize = static_cast<std::size_t>(dynsymHdr->sh_size);
-    char* symbols = new char[sectionSize];
-    fs.seekg(dynsymHdr->sh_offset);
-    fs.read(symbols, sectionSize);
-    Sym* symEntry = reinterpret_cast<Sym*>(symbols);
-    for (std::size_t symIndex = 0;
-         symIndex < sectionSize / dynsymHdr->sh_entsize;
-         ++symIndex, ++symEntry) {
-      if (symEntry->st_shndx == SHN_UNDEF ||
-          this->GetSymbolEntryType(symEntry->st_info) != STT_FUNC) {
-        continue;
-      }
-      std::string symName = strTab + symEntry->st_name;
-      if (this->ExtractBundleName(symName, m_BundleName)) {
-        break;
+    // parse the .us_resources section
+    char* buffer = static_cast<char*>(malloc(m_SectionHeaders[m_FileHeader.e_shstrndx].sh_size));
+    if(nullptr != buffer) {
+      // keep the buffer in a smart pointer to correctly clean up at end
+      // of scope even if exceptions occur.
+      std::unique_ptr<void, void(*)(void*)> scopedBuffer(buffer, ::free);
+      fs.seekg(m_SectionHeaders[m_FileHeader.e_shstrndx].sh_offset);
+      fs.read(buffer, m_SectionHeaders[m_FileHeader.e_shstrndx].sh_size);
+    
+      const char* const sh_str = buffer;
+      for (int i = 0; i < m_FileHeader.e_shnum; ++i)
+      {
+        if (0 == strcmp(".us_resources", (sh_str + m_SectionHeaders[i].sh_name)))
+        {
+          fs.seekg(m_SectionHeaders[i].sh_offset);
+          auto zipContentSize = m_SectionHeaders[i].sh_size;
+          if(0 < zipContentSize) {
+            auto zipContent = static_cast<char*>(malloc(zipContentSize));
+            if (zipContent) {
+              std::unique_ptr<void, void(*)(void*)> scopedData(zipContent, ::free);
+              fs.read(reinterpret_cast<char*>(zipContent), zipContentSize);
+              m_rawData = std::make_shared<RawBundleResources>(std::move(scopedData), zipContentSize);
+              break;
+            }
+          }
+        }
       }
     }
-    delete[] symbols;
+  }
 
+  ~BundleElfFile()
+  {
     for (typename StrTblMapType::const_iterator
            iter = m_StrTblIndexToStrArray.begin(),
            iterEnd = m_StrTblIndexToStrArray.end();
          iter != iterEnd;
          ++iter) {
       delete[] iter->second;
     }
-
-    delete[] m_SectionHeaders;
   }
 
   virtual std::vector<std::string> GetDependencies() const { return m_Needed; }
```

Comment:
Omit `virtual` and add `override`?

(this comment for all instances of where such a change applies)